### PR TITLE
CLN: OrderedDict -> Dict

### PR DESCRIPTION
**Pull request by alimcmaster1**
- xref https://github.com/pandas-dev/pandas/issues/30469
- [x] follow up from https://github.com/pandas-dev/pandas/pull/29212
- [x] passes `black pandas`

As of CPython 3.6 the implementation of dict maintains insertion order.

This became a language feature in python 3.7.

Note in python 3.8 support for reversed(dict) was added - but we don't seem to use that in the use cases i've eliminated.

There is more to do here.

### CODE REVIEW

**Comment by topper-123** [comment]
Can you use dict literals where possible, i.e. ``{"": (0, 0)} `` ``dict((("", (0, 0)),))`` etc.

Else LGTM.

**Review by WillAyd — APPROVED** [review]
lgtm. Should maybe add a code check as well to not use OrderedDict, or is there any outstanding reason still to use?

@jreback

**Comment by jreback** [comment]
@alimcmaster1 yeah lgtm. but agree we need a code check here, can you add.

**Comment by alimcmaster1** [comment]
Thanks @topper-123 updated as per your comments. @jreback sure will add a code check in a follow up - there are still some more references to remove but ill save those for next PR

File: pandas/io/stata.py [hunk 1]
```diff
@@ -2992,7 +2991,7 @@ def _write_map(self):
         the map with 0s.  The second call writes the final map locations when
         all blocks have been written."""
         if self._map is None:
-            self._map = OrderedDict(
+            self._map = dict(
                 (
```
Comment (topper-123) [inline pandas/io/stata.py]: A last one missing here, else looks good.

**Comment by WillAyd** [comment]
Thanks @alimcmaster1

**Pull request merged**